### PR TITLE
Migrate to Capacitor 4

### PR DESCRIPTION
**Pull request by Nevvulo**
Updates the plugin to support Capacitor 4.
I've confirmed this is working for Android and iOS, tested in the simulator and through `npm run verify`

## Capacitor 3 vs Capacitor 4

- Version 0.1.3 (and lower) will target the latest Capacitor version
- Version 1.0.0 will target Capacitor 4

## Breaking Changes
* Plugin must be imported like `import { Segment } from '@joinflux/capacitor-segment'` (instead of `CapacitorSegment`)
* Version 1.x.x will solely support Capacitor 4
* No functionality has changed

### CODE REVIEW

File: android/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="com.joinflux.flux.capacitorsegment">
```
Comment (anonimitoraf) [inline android/src/main/AndroidManifest.xml:2]: Just wondering why the package name has changed?

File: android/src/main/java/com/joinflux/flux/segment/SegmentPlugin.java [hunk 2]
```diff
@@ -26,24 +26,27 @@ public void initialize(PluginCall call) {
...
         }
         String key = call.getString("key");
+        Boolean trackLifecycle = call.getBoolean("trackLifecycle", false);
         if (key == null) {
             call.reject("Write key is required to initialize plugin");
             return;
         }
 
         Context context = this.getContext();
-        Builder builder = new Analytics.Builder(context, key);
```
Comment (anonimitoraf) [inline android/src/main/java/com/joinflux/flux/segment/SegmentPlugin.java:35]: Why was this code moved back into the UI thread? (i.e. this [commit](https://github.com/joinflux/capacitor-segment/commit/62555b325a770bc77d70e3c4ec7f98797b8434b0) seems to have been reverted)?

**Review by anonimitoraf — COMMENTED** [review]
Nice job on this :) 

I just left a couple of questions

File: android/src/main/java/com/joinflux/flux/segment/SegmentPlugin.java (hunk 2 above)
Comment (Nevvulo) [reply android/src/main/java/com/joinflux/flux/segment/SegmentPlugin.java:35]: Thanks for picking this up, this was not meant to be re-added, will fix this

File: android/gradlew [hunk 3]
```diff
@@ -1,7 +1,7 @@
-#!/usr/bin/env sh
```
Comment (anonimitoraf) [inline android/gradlew:1]: I should've asked this earlier: does gradlew get auto-generated?
I wonder why the shebang changed sh's path. The latter is a bit less robust AFAIK (some systems don't have sh under `/bin`)

Comment (Nevvulo) [reply android/gradlew:1]: It does get automatically generated when creating a new plugin, which is what I've done for migrating to Capacitor 4. I'm not sure why this has changed but it hasn't caused any issues for me personally while building or testing the updated plugins in-app

Comment (anonimitoraf) [reply android/gradlew:1]: Ah ok I see. And yeah, most (POSIX) systems would have /bin/sh but it's encouraged best practice to use /usr/bin/env

**Review by anonimitoraf — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -23,15 +23,20 @@ and
 npx cap sync
 ```
 
+### Capacitor 3 vs Capacitor 4
+
+- Version 0.1.3 will target the latest Capacitor version
```
Comment (gugahoi) [inline README.md:28]: Should we specify that we recommend Capacitor 3 for this version?

File: android/build.gradle [hunk 5]
```diff
@@ -36,22 +36,21 @@ android {
         abortOnError false
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
```
Comment (gugahoi) [inline android/build.gradle:40]: Is this correct? I believe 1 of them is OpenJDK and the other is Oracle JDK.

File: android/src/main/java/com/joinflux/flux/segment/SegmentPlugin.java [hunk 6]
```diff
@@ -26,24 +26,27 @@ public void initialize(PluginCall call) {
...
+                    public void run() {
+                        Builder builder = new Analytics.Builder(context, key);
+                        if (trackLifecycle) {
+                            builder.trackApplicationLifecycleEvents();
+                        }
+                        initialized = true;
+                        implementation.analytics = builder.build();
+                    }
+                }
+            );
```
Comment (gugahoi) [inline android/src/main/java/com/joinflux/flux/segment/SegmentPlugin.java]: Assuming this was copied from Segment documentation? If not, do we have good reason for this change?

**Review by gugahoi — COMMENTED** [review]
Looks mostly good, I have a couple of questions.

File: android/src/main/java/com/joinflux/flux/segment/SegmentPlugin.java (hunk 6 above)
Comment (Nevvulo) [reply android/src/main/java/com/joinflux/flux/segment/SegmentPlugin.java]: This was a mistake, I've reverted this back to the way we had it before (without the need for UI thread). It should be fixed in [73c7b79](https://github.com/joinflux/capacitor-segment/pull/14/commits/73c7b79a7bf6deeac53eba4a7f6691b1247d4e98)

File: android/build.gradle (hunk 5 above)
Comment (Nevvulo) [reply android/build.gradle:40]: This is directly from a fresh Capacitor 4 plugin template from `npx @capacitor/create-plugin`, see [here](https://github.com/ionic-team/create-capacitor-plugin/blob/03027bf603626ebfdba084cb9d9a1a359d008554/assets/plugin-template/android/build.gradle#L39-L40). I haven't had any issues when testing however I did need to update my Android Studio to use the updated version mentioned 
image

File: README.md (hunk 4 above)
Comment (Nevvulo) [reply README.md:28]: Sure, I'll update this

File: android/build.gradle (hunk 5 above)
Comment (gugahoi) [reply android/build.gradle:40]: I see, no worries.

**Comment by Nevvulo** [comment]
Planning to merge this shortly as I believe I've addressed concerns. I've tested the plugin on iOS + Android (simulator + real device) and everything worked as expected

**Review by gugahoi — APPROVED** [review]

**Pull request merged**